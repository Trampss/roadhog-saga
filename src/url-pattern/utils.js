export const encodeParams = (params) => {
  if (!Array.isArray(params)) return encodeURIComponent(params)
  return params.map(param => encodeURIComponent(param))
}

export const isEmpty = (o) => {
  return !o ||
    ((typeof o === 'object' && Object.keys(o).length === 0) ||
    (Array.isArray(o) && o.length === 0))
}
