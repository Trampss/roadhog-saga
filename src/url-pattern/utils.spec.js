/* eslint-env jest */
import { encodeParams, isEmpty } from './utils'

describe('utils', () => {
  describe('encodeParams', () => {
    it('should encode params (one param)', () => {
      expect(encodeParams('a param')).toMatchSnapshot()
    })

    it('should encode params (array params)', () => {
      expect(encodeParams(['a param', 'an other :)'])).toMatchSnapshot()
    })
  })

  describe('isEmpty', () => {
    it('should be empty when undefined', () => {
      expect(isEmpty(undefined)).toMatchSnapshot()
    })

    describe('object', () => {
      it('should be empty', () => {
        expect(isEmpty({})).toMatchSnapshot()
      })

      it('should not be empty', () => {
        expect(isEmpty({ some: 'field' })).toMatchSnapshot()
      })
    })

    describe('array', () => {
      it('should be empty', () => {
        expect(isEmpty([])).toMatchSnapshot()
      })

      it('should not be empty', () => {
        expect(isEmpty(['some'])).toMatchSnapshot()
      })
    })
  })
})
