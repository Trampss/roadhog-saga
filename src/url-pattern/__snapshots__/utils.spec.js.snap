exports[`utils encodeParams should encode params (array params) 1`] = `
Array [
  "a%20param",
  "an%20other%20%3A)",
]
`;

exports[`utils encodeParams should encode params (one param) 1`] = `"a%20param"`;

exports[`utils isEmpty array should be empty 1`] = `true`;

exports[`utils isEmpty array should not be empty 1`] = `false`;

exports[`utils isEmpty object should be empty 1`] = `true`;

exports[`utils isEmpty object should not be empty 1`] = `false`;

exports[`utils isEmpty should be empty when undefined 1`] = `true`;
